membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -52,13 +55,24 @@ class AttributesFromZuora(implicit val executionContext: ExecutionContext, syste
           case Failure(e) => SafeLogger.error(scrub"Failed to retrieve attributes from cache for $identityId because $e")
         }
 
+    def userHasDigiSub(accountsWithSubscriptions: List[AccountWithSubscriptions]) =
+      accountsWithSubscriptions.flatMap(_.subscriptions).exists(_.asDigipack.isDefined)
+
     lazy val getAttrFromZuora =
       for {
         accounts <- EitherT(identityIdToAccounts(identityId))
-        subscriptions <- EitherT(if (accounts.size > 0) getSubscriptions(accounts, subscriptionsForAccountId) else Future.successful(\/.right[String, List[AccountWithSubscriptions]](Nil)))
+        subscriptions <- EitherT(
+          if (accounts.size > 0) getSubscriptions(accounts, subscriptionsForAccountId)
+          else Future.successful(\/.right[String, List[AccountWithSubscriptions]](Nil))
+        )
+        giftSubscriptions <- EitherT(
+          if(userHasDigiSub(subscriptions)) Future.successful(\/.right[String, List[GiftSubscriptionsFromIdentityIdRecord]](Nil))
```

Comment:
I can't tell where `subscriptions` has cancelled subs or old fixed terms filtered out. It's not in the query https://github.com/guardian/membership-common/blob/7d9d4ab369d59a8c918f1469d2f32723f34ab0ef/src/main/scala/com/gu/zuora/rest/ZuoraRestService.scala#L375-L378 and I can't see it in Zuora attributes maker.  So I assume it's buried in the anyPlanReads so it probably only does Cancelled subs and not fixed terms that have ended.
As far as I know we never did fixed term digi subs and paper+digital.

Reply:
I'm filtering out gifts from this list but we probably need to filter expired corporate subs as well. I've added a card to Trello: https://trello.com/c/AO59JiJX/3138-make-sure-that-expired-corporate-subs-no-longer-give-benefits

Reply:
thanks, I don't know if it's safer to just have a general TermEndDate check somewhere